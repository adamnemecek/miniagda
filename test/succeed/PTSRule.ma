-- 2010-09-22

-- a buggy PTS rule might check whether the sort of the domain
-- is leq than the sort of the codomain

let T : (i : Size) -> Set ($$ i)
  = \ i -> Set ($ i) -> Set i

let U : (i : Size) -> Set _
  = \ i -> Set ($ _) -> Set _

