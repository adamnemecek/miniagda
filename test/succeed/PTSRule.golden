--- opening "PTSRule.ma" ---
--- scope checking ---
--- type checking ---
mixk  T : (i : Size) -> Set $$i
mixk  T = \ i -> Set $i -> Set i
mixk  U : (i : Size) -> Set ?0
mixk  U = \ i -> Set $?1 -> Set ?2
--- evaluating ---
--- closing "PTSRule.ma" ---
